Source/Core/Common/GL/GLInterface/EGL.cpp

```diff
@@ -98,15 +100,43 @@ void cInterfaceEGL::DetectMode()
     delete[] config;
   }
 
-  if (supportsGL)
-    s_opengl_mode = GLInterfaceMode::MODE_OPENGL;
-  else if (supportsGLES3)
-    s_opengl_mode = GLInterfaceMode::MODE_OPENGLES3;
-  else if (supportsGLES2)
-    s_opengl_mode = GLInterfaceMode::MODE_OPENGLES2;
+  if (preferGLES)
+  {
+    if (supportsGLES3)
+      s_opengl_mode = GLInterfaceMode::MODE_OPENGLES3;
+    else if (supportsGLES2)
+      s_opengl_mode = GLInterfaceMode::MODE_OPENGLES2;
+    else if (supportsGL)
+      s_opengl_mode = GLInterfaceMode::MODE_OPENGL;
+  }
+  else
+  {
+    if (supportsGL)
+      s_opengl_mode = GLInterfaceMode::MODE_OPENGL;
+    else if (supportsGLES3)
+      s_opengl_mode = GLInterfaceMode::MODE_OPENGLES3;
+    else if (supportsGLES2)
+      s_opengl_mode = GLInterfaceMode::MODE_OPENGLES2;
+  }
 
-  if (s_opengl_mode == GLInterfaceMode::MODE_DETECT)  // Errored before we found a mode
-    s_opengl_mode = GLInterfaceMode::MODE_OPENGL;     // Fall back to OpenGL
+  if (s_opengl_mode == GLInterfaceMode::MODE_OPENGL)
+  {
+    INFO_LOG(VIDEO, "Using OpenGL");
+  }
+  else if (s_opengl_mode == GLInterfaceMode::MODE_OPENGLES3)
+  {
+    INFO_LOG(VIDEO, "Using OpenGL|ES 3");
+  }
+  else if (s_opengl_mode == GLInterfaceMode::MODE_OPENGLES2)
+  {
+    INFO_LOG(VIDEO, "Using OpenGL|ES 2");
+  }
+  else if (s_opengl_mode == GLInterfaceMode::MODE_DETECT)
+  {
+    // Errored before we found a mode
+    INFO_LOG(VIDEO, "Error: Failed to detect OpenGL flavour, falling back to OpenGL");
+    s_opengl_mode = GLInterfaceMode::MODE_OPENGL;  // Fall back to OpenGL
```

Comment:
This can only be hit if there was no EGL_RENDERABLE_TYPE that created a valid config, which *probably* should be a fatal error (or at least a "louder" error) as I can't think of any sane way of continuing as the "default" for that is EGL_OPENGL_ES_BIT, and we clearly will never support ES1